aptos-move/framework/src/built_package.rs

```diff
@@ -237,7 +237,45 @@ impl BuiltPackage {
     pub fn build(package_path: PathBuf, options: BuildOptions) -> anyhow::Result<Self> {
         let build_config = Self::create_build_config(&options)?;
         let resolved_graph = Self::prepare_resolution_graph(package_path, build_config.clone())?;
-        BuiltPackage::build_with_external_checks(resolved_graph, options, build_config, vec![])
+        BuiltPackage::build_with_external_checks_impl(
+            resolved_graph,
+            options,
+            build_config,
+            false,
+            vec![],
+        )
+    }
+
+    /// Same as `build` but allows to provide external checks to be made on Move code.
+    /// The `external_checks` are only run when compiler v2 is used.
+    pub fn build_with_external_checks(
+        resolved_graph: ResolvedGraph,
+        options: BuildOptions,
+        build_config: BuildConfig,
+        external_checks: Vec<Arc<dyn ExternalChecks>>,
+    ) -> anyhow::Result<Self> {
+        Self::build_with_external_checks_impl(
+            resolved_graph,
+            options,
+            build_config,
+            false,
+            external_checks,
+        )
+    }
+
+    /// Used only for testing to skip errors (e.g., when running extended checks) when compiling
+    /// and building a package.
+    #[cfg(any(test, feature = "testing"))]
+    pub fn build_skip_errors(package_path: PathBuf, options: BuildOptions) -> anyhow::Result<Self> {
```

Comment:
Why do we want to skip errors? To check if the VM handles them correctly if compiler was bypassed?

If so, it would be better to do this as an experiment (say `Experiment::SKIP_BAILOUT_ON_EXTENDED_CHECKS` or similar), and pass the experiment at the call site (via `options`), instead of threading through another parameter. The experiment setup was created to minimize adding all these parameters.

Reply:
Yes, to check the VM catches these errors.

Good to know about it, added an experiment as you suggest. Btw, this means experiment is available in CLI, do we want to have `#[cfg(test)]` or similar configs to enable experiments only for testing purposes?

Reply:
The experiment flag is hidden in the CLI, so it does not show up when you do `aptos move compile --help`.

Ofcourse, a user can look at the source code, and pass in appropriate experiments. But in such a case, they could also modify the source to disable corresponding checks.

Reply:
Makes sense!